release/models/aft/openconfig-aft-common.yang

```diff
@@ -385,6 +407,37 @@ submodule openconfig-aft-common {
     }
   }
 
+  grouping aft-common-entry-nexthop-gre-state {
+    description
+      "GRE encapsulation applied on a next-hop";
+
+    leaf src-ip {
+      type union {
+        type oc-inet:ip-address;
+        type string;
+      }
+      description
+        "Source IP address to use for the encapsulated packet. This
+         may be expressed as either an IP address or reference
+         to the name of an interface.";
+    }
+
+    leaf dst-ip {
+      type oc-inet:ip-address;
```

Comment:
Is it possible to set this to be a link-local address for a v6 destination such that we should use the `-zoned` type?

Reply:
Maybe I didnt understand your question correctly. Please feel free to elaborate.

My understanding of the GRE encapsulation is that source and destination IP address for the GRE header and the network designer needs to make sure the destination is reachable in the underlay network. The GRE encapsulation itself does not care about the `link-local` address as long as the routing table resolves the GRE packet correctly.

Reply:
agreed -- so if we wanted to set the destination IP to be something that was link local -- because the routing table _could_ resolve this, then we might want to be able to say `fe80::1%eth0` here, if this is the case then we should use [`oc-inet:ip-address-zoned`](https://github.com/openconfig/public/blob/master/release/models/types/openconfig-inet-types.yang#L296) which allows us to specify that.

i'm torn a little on whether this is needed, because we also have `interface-ref` which specifies the egress interface, so in this case, it might be redundant information.

Reply:
@robshakir my understanding from your other comment ( `Given that this is state, can we not just simplify this down to only using the IP address? The implementation can resolve what address is being used if needs be.` ) was we want to not have `interface-ref` and only `ip-address`

Let me know if it still makes sense to keep it as `ip-address` or make it [oc-inet:ip-address-zoned](https://github.com/openconfig/public/blob/master/release/models/types/openconfig-inet-types.yang#L296)

Reply:
@robshakir I agree that with respect to `dst-ip` this case is redundant given `interface-ref`

I am not very familiar with `link-local` addresses but I would expect this `dst-ip` to be the destination IP address programmed in the GRE header as the state. [This](https://github.com/openconfig/public/pull/1038#discussion_r1467855575) comment of yours from earlier implies the implementation can resolve what address is being used if needs be.

But I might be missing context given by limited knowledge of  `link-local` addresses

Reply:
If we clearly specify that use of a link-local address also requires specifying an interface-ref then I think it's redundant to using zoned-ip's. (which to save others from looking this up, a zoned IP merely concatenates an ip-address and interface name together).

Reply:
So in short, I think a zoned IP is not needed (redundant) because we have 
`network-instances/network-instance/afts/next-hops/next-hop/gre/state/dst-ip`

and these (preexisting paths):
`network-instances/network-instance/afts/next-hops/next-hop/interface-ref/state/interface`
`network-instances/network-instance/afts/next-hops/next-hop/interface-ref/state/subinterface`